From f8e2c057ec6123b0a1d08294f41e78c427636b49 Mon Sep 17 00:00:00 2001
From: Mike Bjorge <mbjorge@chromium.org>
Date: Mon, 7 Aug 2017 18:41:58 +0000
Subject: [PATCH] Add mojo_blink dep from WebKit loader/.

Now that loader has a direct dependency on the generated mojom-blink
headers (https://chromium-review.googlesource.com/c/577488), the loader/
target should have a dep on the mojo target that generates the headers,
otherwise the build may fail depending on build order.

All the other platform/ targets that depend on the mojom-blink headers
are directly included in the platform target, which has the necessary
dependency already.

Bug: 746197
Bug: 750883
Change-Id: Ife1429ec649605aae3f7920560cec37f3a218154
Reviewed-on: https://chromium-review.googlesource.com/602869
Reviewed-by: Yutaka Hirano <yhirano@chromium.org>
Reviewed-by: Matt Falkenhagen <falken@chromium.org>
Commit-Queue: Mike Bjorge <mbjorge@chromium.org>
Cr-Commit-Position: refs/heads/master@{#492369}
---
 third_party/WebKit/Source/platform/loader/BUILD.gn | 1 +
 third_party/WebKit/public/BUILD.gn                 | 1 +
 2 files changed, 2 insertions(+)

diff --git a/third_party/WebKit/Source/platform/loader/BUILD.gn b/third_party/WebKit/Source/platform/loader/BUILD.gn
index d483734b21cf..409befe884a8 100644
--- a/third_party/WebKit/Source/platform/loader/BUILD.gn
+++ b/third_party/WebKit/Source/platform/loader/BUILD.gn
@@ -99,6 +99,7 @@ source_set("loader") {
     "//components/link_header_util:link_header_util",
     "//mojo/public/cpp/system:system",
     "//storage/public/interfaces:interfaces_blink__generator",
+    "//third_party/WebKit/public:mojo_bindings_blink",
   ]
 
   public_deps = [
diff --git a/third_party/WebKit/public/BUILD.gn b/third_party/WebKit/public/BUILD.gn
index 84ee3ca9136d..cc847663efc6 100644
--- a/third_party/WebKit/public/BUILD.gn
+++ b/third_party/WebKit/public/BUILD.gn
@@ -755,6 +755,7 @@ mojom("mojo_bindings") {
   visibility_blink = [
     "//content/common:mojo_bindings_blink",
     "//third_party/WebKit/Source/platform",
+    "//third_party/WebKit/Source/platform/loader",
   ]
   sources = [
     "platform/display_mode.mojom",
-- 
2.14.0

